internal/consul/namespaces.go

```diff
@@ -0,0 +1,50 @@
+package consul
+
+import (
+	capi "github.com/hashicorp/consul/api"
+)
+
+const (
+	WildcardNamespace = "*"
+	DefaultNamespace  = "default"
+)
+
+// EnsureNamespaceExists ensures a Consul namespace with name ns exists. If it doesn't,
+// it will create it and set crossNSACLPolicy as a policy default.
+// Boolean return value indicates if the namespace was created by this call.
+func EnsureNamespaceExists(client *capi.Client, ns string, crossNSAClPolicy string) (bool, error) {
+	if ns == WildcardNamespace || ns == DefaultNamespace {
```

Comment:
Does the namespace named `default` always exists no matter what?

Reply:
I would assume so, this line is replicated directly from consul-k8s, but I don't know 100%